### PR TITLE
Feature/recovery overview screen

### PR DESCRIPTION
**Pull request by n13**
### 🗃 Github Issue Or Explanation for this PR. (What is it supposed to do and Why is needed)

As per design spec

### ✅ Checklist

- [ ] Github issue details are up to date for people to QA.
- [x] I have tested all my changes.

### 🕵️‍♂️ Notes for Code Reviewer

1 - Testing with mock data for this PR

2 - Added a setting for the currently active recovery "lost account" - this data cannot be retrieved from chain.

It can be retrieved when entering the same account to recover again though, meaning there's only ever one active recovery on chain with a specific rescuer / lost account pair. 

So if I am the same rescuer, and enter the same lost account as on another device, I am getting the same active recovery back
We need to still store this value, next PR

### 🙈 Screenshots

Simulator Screen Shot - iPhone 13 - 2022-09-28 at 21 50 33

### CODE REVIEW

File: lib/datasource/local/flutter_js/substrate_service.dart [hunk 1]
```diff
@@ -42,6 +42,7 @@ class SubstrateService {
   }
```
Comment (n13) [inline lib/datasource/local/flutter_js/substrate_service.dart:42]: fix for the recovery which I noticed didn't work - should be in separate PR

File: lib/screens/authentication/recover/recover_account_overview/interactor/components/recover_account_overview_view.dart [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                  SettingsCard(
+                    icon: const Icon(Icons.replay_circle_filled_outlined),
+                    title: "Recover an Account",
+                    description: "Recover an account with the help of the guardians set for that account.",
+                    onTap: () async {
+                      BlocProvider.of<RecoverAccountOverviewBloc>(context).add(const OnRecoverAccountTapped());
+                    },
+                  ),
+                  const SizedBox(height: 16),
+                  if (state.activeRecovery != null)
```
Comment (gguijarro-c-chwy) [inline lib/screens/authentication/recover/recover_account_overview/interactor/components/recover_account_overview_view.dart:46]: if (state.activeRecovery != null) if this is false you end up with two 
    const SizedBox(height: 16),
     const SizedBox(height: 16),
     
     fix: 
     if (state.activeRecovery != null)...[
     SettingsCard(
                        backgroundColor: context.colorScheme.tertiaryContainer,
                        textColor: context.colorScheme.onTertiaryContainer,
                        icon: const Icon(Icons.keyboard_command_key),
                        title: "Recovery in Process",
                        description: "Recovering    ${state.activeRecovery!.lostAccount.shorter}",
                        onTap: () async {
                          BlocProvider.of<RecoverAccountOverviewBloc>(context)
                              .add(OnRecoveryInProcessTapped(state.activeRecovery!.lostAccount));
                        }),
                  const SizedBox(height: 16)
],

File: lib/screens/authentication/recover/recover_account_overview/interactor/usecase/fetch_recover_account_overview_data.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import 'package:hashed/blocs/deeplink/model/guardian_recovery_request_data.dart';
+import 'package:hashed/datasource/local/settings_storage.dart';
+import 'package:hashed/datasource/remote/api/guardians_repository.dart';
+import 'package:hashed/datasource/remote/model/active_recovery_model.dart';
+import 'package:hashed/datasource/remote/polkadot_api/polkadot_repository.dart';
+import 'package:hashed/domain-shared/base_use_case.dart';
+import 'package:hashed/domain-shared/shared_use_cases/cerate_firebase_dynamic_link_use_case.dart';
+
+class FetchRecoverAccountOverviewData {
```
Comment (gguijarro-c-chwy) [inline lib/screens/authentication/recover/recover_account_overview/interactor/usecase/fetch_recover_account_overview_data.dart:9]: is this a use case? If so, it should be named as such

File: lib/screens/settings/components/account_card.dart [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import 'package:flutter/material.dart';
+import 'package:hashed/domain-shared/ui_constants.dart';
+import 'package:hashed/utils/ThemeBuildContext.dart';
+import 'package:hashed/utils/short_string.dart';
+
+class AccountCard extends StatelessWidget {
+  final String address;
+  final String? accountName;
```
Comment (gguijarro-c-chwy) [inline lib/screens/settings/components/account_card.dart:8]: can this ever be null? 
looks to me that an AccountCard should always have an account

**Review by gguijarro-c-chwy — COMMENTED** [review]
Minor suggestions

File: lib/screens/settings/components/account_card.dart (hunk 4 above)
Comment (n13) [reply lib/screens/settings/components/account_card.dart:8]: Yes it always has an address, but we will fetch account names, and they can be null (ie there's no name set on the identity pallet)

File: lib/screens/authentication/recover/recover_account_overview/interactor/usecase/fetch_recover_account_overview_data.dart (hunk 3 above)
Comment (n13) [reply lib/screens/authentication/recover/recover_account_overview/interactor/usecase/fetch_recover_account_overview_data.dart:9]: I know... I copied that code from you ;) changing...

File: lib/screens/authentication/recover/recover_account_overview/interactor/components/recover_account_overview_view.dart (hunk 2 above)
Comment (n13) [reply lib/screens/authentication/recover/recover_account_overview/interactor/components/recover_account_overview_view.dart:46]: Yeah that's unclean. Is there a more elegant solution than to add 2 "if" statements?

Comment (n13) [reply lib/screens/authentication/recover/recover_account_overview/interactor/components/recover_account_overview_view.dart:46]: Fixed with 2 if statements

Comment (gguijarro-c-chwy) [reply lib/screens/authentication/recover/recover_account_overview/interactor/components/recover_account_overview_view.dart:46]: Yes, i pasted the fix in this comment for you lol.

You only need one if statement and (...)

if(Bool)...[

**Comment by n13** [comment]
Oh - that's very cool - making that change and checking in
